### PR TITLE
feat: `Predicate` serialization

### LINKED ISSUE #2493: Serialize related Delete Expression to store in Catalog
# Rationale 
As part of #2396  we need to store delete predicate in the catalog for us to eliminate data when reading and compacting.  After a discussion, we decide to serialize the Delete Expr to protobuf messages/structures before storing to avoid backward incompatibility in the future. 

The Delete Expr is subset of DF Expr and implemented in this PR #2488

### PR DESCRIPTION
**Pull request by crepererum**
Closes #2493.

### CODE REVIEW

File: generated_types/protos/influxdata/iox/catalog/v1/predicate.proto [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  // Inequality (`!=`).
+  OP_NE = 2;
+}
+
+// Scalar value of a certain type.
+message Scalar {
+  oneof value {
+    bool value_bool = 1;
+    int64 value_i64 = 2;
+    uint64 value_u64 = 3;
```
Comment (NGA-TRAN) [inline generated_types/protos/influxdata/iox/catalog/v1/predicate.proto:77]: FYI. DF parser seems not generate u64. Currently, we only generate i64 and f64

File: predicate/src/serialize.rs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            .table(table_name)
+            .timestamp_range(parse_delete_pred.start_time, parse_delete_pred.stop_time);
+
+        for expr in parse_delete_pred.predicate {
+            del_predicate_builder = del_predicate_builder.add_expr(expr);
+        }
+
+        del_predicate_builder.build()
+    }
+}
```
Comment (NGA-TRAN) [inline predicate/src/serialize.rs:263]: Maybe add a negative test that error happens because of unsupported string?

**Review by NGA-TRAN — APPROVED** [review]
LGTM

File: predicate/src/serialize.rs (hunk 2 above)
Comment (crepererum) [reply predicate/src/serialize.rs:335]: done

File: generated_types/protos/influxdata/iox/catalog/v1/predicate.proto (hunk 1 above)
Comment (crepererum) [reply generated_types/protos/influxdata/iox/catalog/v1/predicate.proto]: removed that variant, we can easily add it back later

**Pull request merged**